modules/mcuboot/CMakeLists.txt

```diff
@@ -203,6 +211,58 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
       )
   endfunction()
 
+  function(split)
+    # Splits an elf image into 2 sets of hex and bin files
+    #
+    # Required arguments are:
+    # ELF_FILE_IN
+    # ELF_SECTION_NAMES
+    # HEX_INCLUDE_FILE_OUT
+    # HEX_EXCLUDE_FILE_OUT
+    set(oneValueArgs ELF_FILE_IN HEX_INCLUDE_FILE_OUT HEX_EXCLUDE_FILE_OUT)
+    set(multiValueArgs ELF_SECTION_NAMES DEPENDS)
+    cmake_parse_arguments(SPLIT_ARG "" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+    check_arguments_required_all(split SPLIT_ARG ${oneValueArgs} ELF_SECTION_NAMES)
+
+    set(include_param)
+    set(exclude_param)
+
+    foreach(section_name ${SPLIT_ARG_ELF_SECTION_NAMES})
+      set(include_param ${include_param} -R ${section_name})
+      set(exclude_param ${exclude_param} -j ${section_name})
+    endforeach()
+
+    add_custom_command(
+      OUTPUT
+      ${SPLIT_ARG_HEX_INCLUDE_FILE_OUT}
+
+      COMMAND
+      ${CMAKE_OBJCOPY}
+      --output-target=ihex
+      ${include_param}
+      ${SPLIT_ARG_ELF_FILE_IN}
+      ${SPLIT_ARG_HEX_INCLUDE_FILE_OUT}
+
+      DEPENDS
+      ${SPLIT_ARG_DEPENDS}
```

Comment:
What is this for? I can not find a definition for SPLIT_ARG_DEPENDS nor what that thing does.

Reply:
OK, figured this out.